src/main/java/com/flightstats/hub/spoke/SpokeTtlEnforcer.java

```diff
@@ -27,38 +24,32 @@
 @Slf4j
 public class SpokeTtlEnforcer {
 
-    private final SpokeStore spokeStore;
     private final ChannelService channelService;
     private final SpokeContentDao spokeContentDao;
     private final StatsdReporter statsdReporter;
+    private final SpokeProperties spokeProperties;
 
-    private final String storagePath;
-    private final int ttlMinutes;
+    private SpokeStore spokeStore;
+    private String storagePath;
+    private int ttlMinutes;
 
     @Inject
-    public SpokeTtlEnforcer(SpokeStore spokeStore,
-                            ChannelService channelService,
+    public SpokeTtlEnforcer(ChannelService channelService,
                             SpokeContentDao spokeContentDao,
                             StatsdReporter statsdReporter,
                             SpokeProperties spokeProperties) {
-        this.spokeStore = spokeStore;
         this.channelService = channelService;
         this.spokeContentDao = spokeContentDao;
         this.statsdReporter = statsdReporter;
-
-        this.storagePath = spokeProperties.getPath(spokeStore);
-        this.ttlMinutes = spokeProperties.getTtlMinutes(spokeStore) + 1;
-        if (spokeProperties.isTtlEnforced()) {
-            HubServices.register(new SpokeTtlEnforcerService());
-        }
+        this.spokeProperties = spokeProperties;
     }
 
     private Consumer<ChannelConfig> handleCleanup(AtomicLong evictionCounter) {
         return channel -> {
             int itemsEvicted = 0;
-            String channelPath = storagePath + "/" + channel.getDisplayName();
+            final String channelPath = storagePath + "/" + channel.getDisplayName();
             if (channel.isLive()) {
-                DateTime ttlDateTime = TimeUtil.stable().minusMinutes(ttlMinutes + 1);
+                final DateTime ttlDateTime = TimeUtil.stable().minusMinutes(ttlMinutes + 1);
```

Comment:
I'm not sure I follow why we're not setting ttl minutes in the initializer anymore. It means we could technically call this method before `initSpokeStore` is called. Generally I try to avoid any mutable attributes at the class instance level because of potential bugs like that...instead I'd rather pass those values as arguments if you don't know the values at initialization time.

Reply:
...but in this case, you could probably add private getters for some of these values, since this is injecting `SpokeProperties`.

Reply:
SpokeTtlEnforcer needs to know SpokeStore  type to function. This comes from SpokeTtlEnforcerService at runtime. The spoke properties cannot be initialized in constructor since we spoekstore type is known only at runtime.  I would not worry about this since all the methods in this class is private except for cleanup method which sets spokestore type.

Reply:
I'd probably opt to pass spokeStore around to methods that need it in that case. I've seen enough bugs with mutable state that I'll avoid it whenever I can.